Make the occlusion culling debug draw mode draw a translucent overlay

Calinou opened the pull request:
This allows for easy, depth-aware navigation in the 3D editor while being able to view which meshes are occluders and which meshes aren't.

I've tested this change with both Forward+ and Mobile rendering methods. No Vulkan validation errors are reported when running with `--gpu-validation` on either rendering method. (The Compatibility rendering method currently doesn't support most debug draw modes, including the occlusion culling buffer. Occlusion culling itself does work though.)

Thanks @myaaaaaaaaa for the [help](https://github.com/godotengine/godot-proposals/issues/2911#issuecomment-1579958923) :slightly_smiling_face:

**Testing project:** [test_occlusion_culling_buffer.zip](https://github.com/godotengine/godot/files/12115364/test_occlusion_culling_buffer.zip)

- This closes https://github.com/godotengine/godot-proposals/issues/2911.

## Preview

*The two cubes in the front have occluders, while the cube in the back doesn't.*

![image](https://github.com/user-attachments/assets/665af5de-d264-478e-b859-dc8c1686be7e)

Calinou committed:
Make the occlusion culling debug draw mode draw a translucent overlay

This allows for easy, depth-aware navigation in the 3D editor while
being able to view which meshes are occluders and which meshes aren't.

Co-authored-by: myaaaaaaaaa <103326468+myaaaaaaaaa@users.noreply.github.com>

BlueCube3310 commented:
I think this is a useful change, especially since the jittered buffer makes it very difficult to see the shapes of the objects. Needs a rebase though

Calinou commented:
Rebased and tested again, it works as expected.

The "vignette" effect is also gone (it was already fixed in `master`). I've updated the screenshot in OP accordingly.

BlueCube3310 reviewed (approved):
Looks good to me, I tested it on several scenes with many meshes and it works correctly. Additionally it makes it easier to notice how the occluders interact with the scene

clayjohn commented on servers/rendering/renderer_rd/shaders/effects/copy_to_fb.glsl:
```diff
@@ -203,6 +203,14 @@ void main() {
 		color.rgb = normalize(color.rgb * 2.0 - 1.0) * 0.5 + 0.5;
 	}
 
+#ifdef MODE_OCCLUSION_CULLING_BUFFER
```
This should just be a flag otherwise we are increasing load times for no reason

Calinou replied:
I've pushed a second commit to attempt this, but it doesn't seem to work on my end.

Calinou committed:
Move to flag [ci skip]

Calinou commented on servers/rendering/renderer_rd/effects/copy_effects.cpp:
```diff
@@ -568,6 +586,10 @@ void CopyEffects::copy_to_fb_rect(RID p_source_rd_texture, RID p_dest_framebuffe
 	}
 
+	if (p_occlusion_culling_buffer) {
+		copy.push_constant.flags |= COPY_TO_FB_FLAG_OCCLUSION_CULLING_BUFFER;
```
Thanks 🙂  I've tried this but now I just see a fully red viewport when the debug draw mode is enabled. So there's still an issue somewhere else.

BlueCube3310 replied:
The correct pipeline is never applied since it relies on the "COPY_TO_FB_OCCLUSION_CULLING_BUFFER" mode that doesn't get used now
